drivers/sensor/mpu9250/mpu9250.h

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) 2021, Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_DRIVERS_SENSOR_MPU9250_MPU9250_H_
+#define ZEPHYR_DRIVERS_SENSOR_MPU9250_MPU9250_H_
+
+#include <device.h>
+#include <drivers/gpio.h>
+#include <sys/util.h>
+#include <zephyr/types.h>
+
+#define MPU9250_REG_CHIP_ID		0x75
+#define MPU9250_CHIP_ID			0x71
+
+#define MPU9250_REG_SR_DIV		0x19
+
+#define MPU9250_REG_CONFIG		0x1A
+#define MPU9250_GYRO_DLPF_MAX		7
+
+#define MPU9250_REG_GYRO_CFG		0x1B
+#define MPU9250_GYRO_FS_SHIFT		3
+#define MPU9250_GYRO_FS_MAX		3
+
+#define MPU9250_REG_ACCEL_CFG		0x1C
+#define MPU9250_ACCEL_FS_SHIFT		3
+#define MPU9250_ACCEL_FS_MAX		3
+
+#define MPU9250_REG_ACCEL_CFG2		0x1D
+#define MPU9250_ACCEL_DLPF_MAX		7
+
+#define MPU9250_REG_INT_EN		0x38
+#define MPU9250_DRDY_EN			BIT(0)
+
+#define MPU9250_REG_DATA_START		0x3B
+
+#define MPU9250_REG_PWR_MGMT1		0x6B
+#define MPU9250_SLEEP_EN		BIT(6)
+
+
+/* measured in degrees/sec x10 to avoid floating point */
+static const uint16_t mpu9250_gyro_sensitivity_x10[] = {
+	1310, 655, 328, 164
+};
+
+struct mpu9250_data {
+	int16_t accel_x;
+	int16_t accel_y;
+	int16_t accel_z;
+	uint16_t accel_sensitivity_shift;
+
+	int16_t temp;
+
+	int16_t gyro_x;
+	int16_t gyro_y;
+	int16_t gyro_z;
+	uint16_t gyro_sensitivity_x10;
+
+#ifdef CONFIG_MPU9250_MAGN_EN
+	int16_t magn_x;
+	int16_t magn_scale_x;
+	int16_t magn_y;
+	int16_t magn_scale_y;
+	int16_t magn_z;
+	int16_t magn_scale_z;
+#endif
+
+#ifdef CONFIG_MPU9250_TRIGGER
+	const struct device *dev;
```

Comment:
Move this to `struct mpu9250_config` and use `DEVICE_DT_INST_GET` in the intializer

Reply:
I can't move this to mpu9250 config.
Inside work callback I can't access anything other than driver data, and I need to have access to dev.
I can't set this struct by DEVICE_DT_INST_GET, because the device struct is not yet created.
Is there a way to access device struct, by only having device data?